backend/server/routers/courses.py

```diff
@@ -178,3 +178,35 @@ def unselectCourse(userData: UserData, lockedCourses: list, unselectedCourse: st
     affectedCourses = User(fixUserData(userData.dict())).unselect_course(unselectedCourse, lockedCourses)
 
     return {'affected_courses': affectedCourses}
+
+def coursesUnlockedWhenTaken(userData: UserData, courseToBeTaken: str):
+    """ Returns all courses which are unlocked when given course is taken """
+    
+    ## define the user object with user data
+    user = User(fixUserData(userData.dict()))
+    
+    ## initial state
+    courses_initially_unlocked = getAllUnlocked(userData)
+    
+    ## add course to the user
+    courseToAdd = {courseToBeTaken: (getCourse(courseToBeTaken), None)}
```

Comment:
this expects something in this form:

"COMP1511" : [6, 70] where 6 is the UOC, and 70 is the course mark.
I dont think this will behave as you expect.

Reply:
Are you sure? Looks like the key is a string and the value is a tuple of ints? Am i missing something? 
![image](https://user-images.githubusercontent.com/60857954/155863637-f407b9db-9aed-495d-a4cf-08e6260a402a.png)

Reply:
I think any iterable is acceptable here - the key must be the course name and the iterable must have the first thing be the UOC and the second be the grade

Reply:
Ah okay, will have a look

Reply:
Basically I think the only thing you need to do is:
GetCourse(course) -> getCourse(course)['UOC']